lib/rspec/core/profiler.rb

```diff
@@ -0,0 +1,38 @@
+module RSpec
+  module Core
+    # @private
+    class NoProfiler
+      def example_groups
+        {}
+      end
+    end
```

Comment:
This class is stateless.  Is it worth making it a singleton module and then using `NoProfiler` directly instead of `NoProfiler.new` to avoid the extra object allocation?